text/0745-user-provided-cdk-init-templates.md

```diff
@@ -0,0 +1,457 @@
+├── info.json                 # Template metadata
+└── [language]/               # Language directory
+    ├── bin/                  # The CDK app entry point
+    ├── lib/                  # Contains CDK stack definitions and constructs
+    └── cdk.template.json     # CDK configuration with placeholders
+└── [language]/               # Language directory
+    ├── bin/                  # The CDK app entry point
+    ├── lib/                  # Contains CDK stack definitions and constructs
+    └── cdk.template.json     # CDK configuration with placeholders
+```
+
+##### **Example info.json for TypeScript template**
+
+```
+{
+    "description": "My Organization's CDK Template",
+    // can run cdk init command with these alternate names for same template
+    "aliases": ["org-template", "company"] 
+}
+```
+
+##### **Example /bin/%name%.template.ts for TypeScript template**
+
+```
+#!/usr/bin/env node
+import * as cdk from 'aws-cdk-lib';
+import { %name.PascalCased%Stack } from '../lib/%name%-stack';
+
+const app = new cdk.App();
+new %name.PascalCased%Stack(app, '%stackname%');
+```
+
+##### **Example /lib/%name%-stack.template.ts for TypeScript template**
+
+```
+import { Duration, Stack, StackProps } from 'aws-cdk-lib';
+import * as sns from 'aws-cdk-lib/aws-sns';
+import * as subs from 'aws-cdk-lib/aws-sns-subscriptions';
+import * as sqs from 'aws-cdk-lib/aws-sqs';
+import { Construct } from 'constructs';
+
+export class %name.PascalCased%Stack extends Stack {
+  constructor(scope: Construct, id: string, props?: StackProps) {
+    super(scope, id, props);
+
+    const queue = new sqs.Queue(this, '%name.PascalCased%Queue', {
+      visibilityTimeout: Duration.seconds(300)
+    });
+
+    const topic = new sns.Topic(this, '%name.PascalCased%Topic');
+
+    topic.addSubscription(new subs.SqsSubscription(queue));
+  }
+}
+```
+
+##### **Example /cdk.template.json for  TypeScript template**
+
+```
+{
+  "app": "npx ts-node --prefer-ts-exts bin/%name%.ts",
+  "watch": {
+    "include": ["**"],
+    "exclude": [
+      "README.md",
+      "cdk*.json",
+      "**/*.d.ts",
+      "**/*.js",
+      "tsconfig.json",
+      "package*.json",
+      "yarn.lock",
+      "node_modules",
+      "test"
+    ]
+  }
+}
+```
+
+
+
+#### Using Placeholder or Template Variables for Consistent Naming of Template Components
+
+As described in the above examples, the `cdk init` command supports defining custom template components using the `%name.PascalCased%` placeholder or template variable to dynamically insert user determined template file names into generated CDK project files. Below is a breakdown of placeholder or template variables terminology:
+
+* **%...%:** The percent signs indicate a template token that will be replaced during code generation.
+* **name:** This refers to a variable input which will be the name of your CDK project’s target directory.
+* .**PascalCased:** This is a transformation function that converts the string into PascalCase
```

Comment:
Is this the only place holder we provide? Your examples also make use of %stackName%. This would be a good place to show a table with all possible place holders, and what their value is derived from. 

However, before doing that, lets talk about whether we really want to commit to this as a public contract.